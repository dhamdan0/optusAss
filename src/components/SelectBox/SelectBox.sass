.selectBox
    width: 200px
    padding: 1rem
    font-size: var(--font-size-md)
    color: var(--color-primary-main)
    font-weight: bold
