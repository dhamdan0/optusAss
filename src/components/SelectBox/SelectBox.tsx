import './SelectBox.sass'
import { type ChangeEventHandler } from 'react'

interface SelectOption {
  value: number
  label: string
}

interface SelectBoxProps {
  options: SelectOption[]
  onChangeOption: ChangeEventHandler<HTMLSelectElement>
  selectedValue: SelectOption['value']
}
const cssPrefix: string = 'selectBox'

function SelectBox ({ selectedValue, options, onChangeOption }: SelectBoxProps) {
  if (options?.length === 0) return null

  return (
      <>
        <select name="selectBox"
                value={selectedValue}
                className={cssPrefix}
                onChange={onChangeOption}
        >
          { options?.map((option) =>
              <option value={option.value} key={`${option.label}-${option.value}}`}>
                {option.label}
              </option>
          ) }
        </select>
      </>
  )
}

export default SelectBox
