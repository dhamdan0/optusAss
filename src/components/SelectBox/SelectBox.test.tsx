import React from 'react'
import { render } from '@testing-library/react'
import SelectBox from './SelectBox'

function testComponentWithProps (props?: any) {
  const container = render(<SelectBox
      options={[]}
      onChangeOption={ () => {}}
      selectedValue={0}
      {...props}
  />)
  expect(container).toMatchSnapshot()
}

describe('SelectBox', () => {
  it('renders the component when there are no options', () => {
    testComponentWithProps()
  })

  it('renders the component when there are options & wrong selected option', () => {
    testComponentWithProps({
      options: [{
        label: 'test 1',
        value: 1
      }],
      selectedValue: -1
    })
  })

  it('renders the component with happy case', () => {
    testComponentWithProps({
      options: [{
        label: 'test 1',
        value: 1
      }],
      selectedValue: 1
    })
  })
})
