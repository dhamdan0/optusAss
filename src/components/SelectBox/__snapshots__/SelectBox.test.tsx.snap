// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox renders the component when there are no options 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SelectBox renders the component when there are options & wrong selected option 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <select
        class="selectBox"
        name="selectBox"
      >
        <option
          value="1"
        >
          test 1
        </option>
      </select>
    </div>
  </body>,
  "container": <div>
    <select
      class="selectBox"
      name="selectBox"
    >
      <option
        value="1"
      >
        test 1
      </option>
    </select>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SelectBox renders the component with happy case 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <select
        class="selectBox"
        name="selectBox"
      >
        <option
          value="1"
        >
          test 1
        </option>
      </select>
    </div>
  </body>,
  "container": <div>
    <select
      class="selectBox"
      name="selectBox"
    >
      <option
        value="1"
      >
        test 1
      </option>
    </select>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
